### PR TITLE
fix: redundant RCN false positive to build SpotBugs with Java 11

### LINKED ISSUE #259: false positive RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE on try-with-resources
FindBugs (eclipse plugin v3.0.1) reports a redundant null check on the closing brace of the following try-with-resources statement:
  
    public void falsePositive() {
      try (X x = new X(); X y = new X()) {
        throw new IllegalStateException();
      } // Redundant nullcheck of x, which is known to be non-null in test.Bug1169.falsePositive()
    }
  
    private static class X implements AutoCloseable {
      @Override
      public void close() {
      }
    }

The finding seems to be reported when more than one AutoClosable is used and the try block ends with a thrown exception.

### PR DESCRIPTION
**Pull request by KengoTODA**
To reproduce bugs with recent Java and its compilers, it's nice to make this project ready to build with them.
Java 14 is the latest version, but 11 is the latest LTS so this PR uses Java 11 to build SpotBugs. refs #1244.
Gradle uses `--release` option to make sure the version of `.class` files are Java 8, so this change won't affect users.

At this moment we won't introduce `module-info.java` because [some of our submodules provide the same package and it's not supported by JPMS](https://github.com/spotbugs/spotbugs/issues/1244#issuecomment-668872019).

## Fix for RCN false positive

To pass the build with Java 11, this PR fixes #259 and #756 by ignoring `RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE` next to the end of try-block which catches `java/lang/Throwable`. See 7700cfc for detail.
This will fix reported false positives, but introduce false negatives in the following case, because we have no way to judge the bytecode to handle exception is generated by `javac` or not.

```java
@NonNull
SomethingClosable s = ...;
try {
  ...
} catch (Throwable t) { // when the catch clause catches Throwable instance,
  // and when we have null check at the beginning of catch block,
  if (s != null) { // this is redundant null check but cannot be detected by SpotBugs any more
    ...
  }
}
```

I hope this false negative is acceptable to us. The redundant null check itself has less impact on runtime, I believe.

### CODE REVIEW

File: spotbugsTestCases/build.gradle [hunk 1]
```diff
@@ -32,6 +32,11 @@ dependencies {
 tasks.withType(JavaCompile).all {
   options.compilerArgs << '-Xlint:none'
   options.encoding = 'UTF-8'
+  if (it.name == 'classesJava11') {
+    options.compilerArgs.addAll(['--release', '11'])
+  } else if (it.name != 'comoileJava') {
```
Comment (danielfn) [inline spotbugsTestCases/build.gradle:37]: Typo

Comment (KengoTODA) [reply spotbugsTestCases/build.gradle:37]: Nice catch! Thanks

**Comment by KengoTODA** [comment]
Found that not #996 but #259 is the root cause. 

Problem is that the java compiler adds redundant null-check in the generated code:
https://gist.github.com/KengoTODA/8011f3860ee3d66397b9b0e247ba82c4

**Comment by KengoTODA** [comment]
@spotbugs/everyone please review this PR when you have time. Thanks in advance!

**Review by henrik242 — APPROVED** [review]
lgtm.

**Comment by jscancella** [comment]
LGTM

**Review by wreulicke — APPROVED** [review]

**Pull request merged**